### DISCUSSION TITLE
Is it expected to perform poorly on audios where multiple languages are spoken (like learn a new language type of audios)?

### DISCUSSION THREAD

**Discussion by nac3r** [openai/whisper#665] [Q&A]
Think of audios that teach you a new language, like Spanish for English speakers. The speaker will speak in both English and Spanish. 

It seems that Whisper does poorly on this circumstance. The detected language will be the first to be spoken in the video. If the video starts in english then english will be transcribed correctly and Spanish will be redacted as [SPEAKING SPANISH].

If I force the language to be Spanish, then the English parts are translated for some parts of the video, transcribed correctly for others, while others completely loss any sense and Whisper will repeat the same phrase until the end of the audio.

Is this the expected behaviour? No fix until new versions or new models appear?

**Comment by guillermo-gabrielli-fer** [comment]
Try using this version https://github.com/openai/whisper/discussions/397 with one of the VAD options if there is a pause between the English and Spanish segments. It will be slower as it needs to deal with smaller segments.

If you are a developer another option is using NeMo with this English-Spanish [model](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/nemo/models/stt_enes_conformer_transducer_large_codesw) but you need to write code and you won't get subtitles out of the box.
